src/util/requestHandler.ts

```diff
@@ -102,18 +102,18 @@ export class RequestHandler {
             if (Array.isArray(file)) {
               data = new MultipartData();
               headers['Content-Type'] = 'multipart/form-data; boundary=' + data.boundary;
-              file.forEach((f) => {
+              file.forEach((f, index) => {
                 if (!f.file) {
                   return;
                 }
-                (data as MultipartData).attach(f.name, f.file, f.name);
+                (data as MultipartData).attach(`files[${f.id ?? index}]`, f.file, f.name);
```

Comment:
The association between `MessageFile` and `MessageAttachmentOptions` differs for the intention of the request itself. Indicating the intention to keep a file for the PATCH request currently has to be done through `file: % | %[]` - but currently does not have anyway to handle existing files or removal of files without associating the file buffer with the mulitpart request.

> tl;dr - As the code currently is, the request has to have the file buffer to perform the request (regardless of the intention to keep the files from the message state before, or not).

Reply:
By all accounts, L106 above should be designed to handle that - but the interface forces the use of a file buffer (even as a file array).